tests/app/authentication/test_database_session_manager.py

```diff
@@ -1,49 +1,124 @@
 import unittest
 
+import app
+from app.data_model.database import EQSession
+from app.authentication.session_storage import SessionStorage, EQ_SESSION_ID
+
+import flask
 from mock import patch, Mock
 from sqlalchemy.exc import IntegrityError
 
-from app.authentication.session_storage import SessionStorage, EQ_SESSION_ID
-
+# SQLAlchemy confuses pylint and we want to override a protected method here
+# pylint: disable=no-member,protected-access
 
 class TestSessionManager(unittest.TestCase):
 
-    @staticmethod
-    def test_store_user_id_rollback():
-        # Given
-        session_storage = SessionStorage()
-        user_id = "1"
+    def setUp(self):
+        self.flask_app = app.create_app()
+        self.session_storage = SessionStorage()
+
+        # Create a patched db_session so we don't need a real database to test against
+        self.db_session_patcher = patch('app.authentication.session_storage.db_session', autospec=True)
+        self.addCleanup(self.db_session_patcher.stop)
+        self.db_session = self.db_session_patcher.start()
+
+        # Note each test will need to create a test_request_context
+        # in order to access the Flask session object
+
+    def test_store_new_user_id_then_replace(self):
+        with self.flask_app.test_request_context('/status'):
+            # Initially there is no session id present
+            self.assertFalse(EQ_SESSION_ID in flask.session)
```

Comment:
I wasn't sure what the default state of flask.session is on start-up and you can't access it until you have a request context; this could be removed now that we've verified the default behaviour if we're confident it won't ever change?